rpcperms/interceptor.go

```diff
@@ -442,10 +451,10 @@ func (r *InterceptorChain) RegisterMiddleware(mw *MiddlewareHandler) error {
 	defer r.Unlock()
 
 	// The name of the middleware is the unique identifier.
-	registered, ok := r.registeredMiddleware[mw.middlewareName]
+	_, ok := r.registeredMiddlewareNames[mw.middlewareName]
 	if ok {
 		return fmt.Errorf("a middleware with the name '%s' is already "+
-			"registered", registered.middlewareName)
+			"registered", mw.middlewareName)
 	}
 
 	// For now, we only want one middleware per custom caveat name. If we
```

Comment:
The error message below needs to be updated too.

Reply:
good catch!!! thanks 🚀

Reply:
updated!